test/controllers/api/commits_controller_test.rb

```diff
@@ -15,6 +15,16 @@ class CommitsControllerTest < ActionController::TestCase
         assert_response :ok
         assert_json '0.sha', commit.sha
       end
+
+      test "#undeployed returns a list of undeployed commits" do
+        commits = @stack.undeployed_commits.map(&:sha)
```

Comment:
nitpick: you can use `.pluck(:sha)` for that. It doesn't matter much because it's a unit test but...